mediaire_toolbox/queue/daemon.py

```diff
@@ -0,0 +1,98 @@
+import logging
+import traceback
+
+from abc import ABC, abstractmethod
+
+from mediaire_toolbox.queue.redis_wq import RedisWQ
+from mediaire_toolbox.queue import tasks
+
+default_logger = logging.getLogger(__name__)
+
+
+"""
+Common interface for creating operative daemons that consume from one of 
+our queues.
+"""
+
+
+class QueueDaemon(ABC):
+
+    def __init__(self,
+                 input_queue: RedisWQ,
+                 result_queue: RedisWQ,
+                 lease_secs: int,
+                 daemon_name: str,
+                 config: dict):
+        """
+        Parameters
+        ----------
+        
+        input_queue:
+            A Redis queue instance from which we will consume Tasks
+        result_queue:
+            The output queue for the daemon, if applicable
+        lease_secs:
+            Lease timeout in seconds when consuming from the queue
+        daemon_name:
+            A unique identifier for this daemon, will be used for logging
+        config:
+            A configuration dictionary with all the necessary extra parameters
+            for this daemon.
+        """
+        self.input_queue = input_queue
+        self.result_queue = result_queue
+        self.lease_secs = lease_secs
+        self.daemon_name = daemon_name
+        self.config = config
+        self.stopped = False
+
+    @abstractmethod
+    def process_task(self, task):
+        """
+        Business logic to be implemented by the daemon, receiving an already
+        deserialized task here.
+        """
+        pass
+
+    def run_once(self):
+        default_logger.info('Waiting for items from queue {}'.format(
+            self.input_queue._main_q_key))
+        item = self.input_queue.lease(
+            lease_secs=self.lease_secs, block=True)
+        try:
+            # TODO Make this class a parameter for better generalization
```

Comment:
The idea I have is that QueueDaemon should be as general as possible. If it can only read DicomTask objects that it can't be used for any other pipeline. Ideally we would make the task type another parameter of the daemon.